### PR TITLE
Create storage file entity

### LINKED ISSUE #247: Create `StorageFile` entity
### Description
Create `StorageFile` entity

### Requirements
- project should be runnable

### Deliverables
- `StorageFile` entity, validation, mappers
- `StorageFile` foundation service, repository and migrations
- `StroageSettings` and `StorageLocationSettings` external configurations
- register all necessary infrastructure to DI

### PR DESCRIPTION
**Pull request by bunyodkamalov**
Closes #247

### CODE REVIEW

File: src/Training.AirBnb.Clone.Backend/AirBnB.Api/Controllers/AccountsController.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@ public class AccountsController(IUserService userService, IMapper mapper) : Cont
     [HttpGet]
     public async ValueTask<IActionResult> Get([FromQuery] FilterPagination filterPagination, CancellationToken cancellationToken)
     {
-        var specification = new QuerySpecification<User>(filterPagination.PageSize, filterPagination.PageToken, true);
+        //var specification = new QuerySpecification<User>(filterPagination.PageSize, filterPagination.PageToken, true);
         var result = await userService.GetAsync(filterPagination.ToQueryPagination(true).ToQuerySpecification(), cancellationToken);
```
Comment (botirjonabdumannonovv) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Api/Controllers/AccountsController.cs:20]: "var specification = new QuerySpecification<User>(filterPagination.PageSize, filterPagination.PageToken, true);"

buni remove qilib yuborsangiz ham bo'ladi menimcha, sonarcloudda xato berishi mumkin

**Review by botirjonabdumannonovv — CHANGES_REQUESTED** [review]

**Review by Xolmuratov02 — DISMISSED** [review]

File: src/Training.AirBnb.Clone.Backend/AirBnB.Api/Mappers/StorageFileMapper.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+using AutoMapper;
+
+namespace AirBnB.Api.Mappers;
+
+public class StorageFileMapper : Profile
+{
+    public StorageFileMapper()
+    {
+        CreateMap<StorageFile, StorageFileDto>().ReverseMap();
+    }
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Api/Mappers/StorageFileMapper.cs:12]: Delete this

File: src/Training.AirBnb.Clone.Backend/AirBnB.Api/Models/DTOs/StorageFileDto.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿using AirBnB.Domain.Entities;
+
+namespace AirBnB.Api.Models.DTOs;
+
+public class StorageFileDto
+{
+    public string FileName { get; set; } = default!;
+}
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Api/Models/DTOs/StorageFileDto.cs:8]: Delete this

File: src/Training.AirBnb.Clone.Backend/AirBnB.Api/Controllers/StorageFileController.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    public async ValueTask<IActionResult> Get([FromQuery] FilterPagination filterPagination, CancellationToken cancellationToken)
+    {
+        var storageFiles = await storageFileService.GetAsync(filterPagination.ToQueryPagination(true).ToQuerySpecification(), cancellationToken);
+        var result = new List<StorageFileDto>();
+        
+        foreach (var file in storageFiles)
+            result.Add(mapper.Map<StorageFileDto>(file));
+        
+        return result.Any() ? Ok(result) : NotFound();
+    }
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Api/Controllers/StorageFileController.cs:24]: Delete this

File: src/Training.AirBnb.Clone.Backend/AirBnB.Api/Data/SeedDataExtensions.cs [hunk 5]
```diff
@@ -40,4 +44,14 @@ private static async ValueTask SeedUsersAsync(this IdentityDbContext dbContext)
...
     }
+
+    private static async ValueTask SeedStorageFilesAsync(this IdentityDbContext dbContext)
+    {
+        var storageFileFaker = new Faker<StorageFile>()
+            .RuleFor(file => file.FileName, data => data.Random.Word())
+            .RuleFor(file => file.Type, StorageFileType.Image);
+
+        await dbContext.AddRangeAsync(storageFileFaker.Generate(10));
+        await dbContext.SaveChangesAsync();
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Api/Data/SeedDataExtensions.cs:55]: Delete this

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by botirjonabdumannonovv — DISMISSED** [review]

**Review by botirjonabdumannonovv — APPROVED** [review]

**Review by Xolmuratov02 — APPROVED** [review]

**Review by abdura52 — APPROVED** [review]

**Review by sulton-max — APPROVED** [review]
Good job

**Pull request merged**